### PR TITLE
Update JDK download link for Mac OS Android set up

### PR DESCRIPTION
**Pull request by leomeloxp**
*First PR to React Native.*

After performing a clean install of macOS Sierra on my machine I tried to get RN set up again and noticed that the link provided for installing Java (required by Android Studio) pointed to the [wrong location](https://www.java.com/en/download/mac_download.jsp). After some time researching I found the right package required by Android Studio [here](http://www.oracle.com/technetwork/java/javase/downloads/jdk8-downloads-2133151.html).

I tried looking for a more general link (as this points to the JDK version 8 and has a weird path: _jdk8-downloads-2133151.html_ - Sadly this was the best I could find.

I also looked through listed issues to try and find a previous attempt to fix this error but couldn't find any references, sorry if I missed anything.

**TL;DR:** JDK download link is incorrect, PR updates it to the correct one.

### CODE REVIEW

**Comment by hramos** [comment]
As this link is for JDK 8, should we update the copy as well to mention that as opposed to 1.8?

I'm also curious if you'd still need to install the Java 1.8 distribution on a clean macOS install.

**Comment by leomeloxp** [comment]
I'm not totally sure if the Java 1.8 install currently mentioned in the Docs is necessary as I ended up installing it before installing the correct JDK. IIRC the JDK include JRE, but don't quote me on it.

Also, I'm not sure why the docs mention 1.8 as the page you're taken to currently also mentions Java 8 update 111.

I'll update the copy to reference Java 8 instead of 1.8.

**Comment by leomeloxp** [comment]
Updated copy to reflect the right version. I double checked with a Java Dev, apparently versions 1.8 and 8 are the same thing. Java used to market their versions as 1.x, now they just use x.

**Comment by hramos** [comment]
Ah, this is confusing. Made a minor edit to clarify 1.8 and 8 are used interchangeably here.

**Comment by hramos** [comment]
@facebook-github-bot shipit
